indexing/src/main/java/com/google/enterprise/cloudsearch/sdk/indexing/Acl.java

```diff
@@ -416,6 +423,36 @@ public static Principal getUserPrincipal(String userId, String identitySourceId)
         .setUserResourceName(String.format(USER_RESOURCE_NAME_FORMAT, identitySourceId, userId));
   }
 
+  private static Supplier<Set<String>> externalGroupNamesSupplier = () -> {
+    try {
+      ExternalGroups groups = ExternalGroups.fromConfiguration();
+      return groups.getExternalGroups().stream()
+      .map(group -> group.getName())
+      .collect(ImmutableSet.toImmutableSet());
+    } catch (IOException e) {
+      return ImmutableSet.of();
```

Comment:
How could this happen? Why not at least log the exception?

Reply:
ExternalGroups.fromConfiguration throws an exception when no groups file is configured, which is probably most of the time.